HW Sockets

UsernameAlexei opened the pull request:
HW Sockets, maximum task

UsernameAlexei committed:
HW Sockets

HW Sockets, maximum task

arybin93 commented on TCP_client.py:
```diff
@@ -0,0 +1,27 @@
+from Data_collection_class import Collection
+from datetime import datetime
+
+sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Context managers as well

UsernameAlexei replied:
Ready, updated.

arybin93 reviewed (commented):
Thank you for your work. Looks good. I've left some comments, please resolve them and update PR.

UsernameAlexei commented on Data_collection_class.py:
```diff
@@ -0,0 +1,42 @@
+        not_pressed = True
+
+        while self.running:
+            keyboard.read_key()
```
Maybe we can start ```stop_collect``` by pressing a specific key? Having previously added the condition to ```stop_collect```
```python    

def start_collect(self):
      self.running = True
      not_pressed = True
        
      keyboard.on_press(self.stop_collect)
      while self.running:
          keyboard.read_key()
          if not_pressed:
              self.counter += 1
              print(self.counter)
              not_pressed = False
          else:
              not_pressed = True

def stop_collect(self, event):
        if event.name =='q':
            self.running = False
            pyautogui.press('esc')
            self.counter -= 1
            threading.Thread.join(self)


```

arybin93 replied:
I mean something like that:
```
def handler(event):
    self.counter += 1

keyboard.on_press(handler) 
while self.running:
    # just waiting small time for saving CPU time
    time.sleep(0.01) 
```
And add `keyboard.unhook_all()` to `stop_collect()` for cancel handling

UsernameAlexei replied:
Updated. Thank you, this is a really very good solution.

UsernameAlexei committed:
Update Data_collection_class.py

Update Data_collection_class.py

UsernameAlexei committed:
Data in CSV format

UsernameAlexei committed:
Update TCP_client.py

Update 
1.Format for sending data time_now, metric_name, value
2.Context managers

UsernameAlexei commented on keystrokes.txt:
```diff
@@ -0,0 +1,5 @@
+['01:28:731867', '9 press the key in 1 min']
```
Thanks for the additional material. Uploaded the updated output keystrokes.csv

UsernameAlexei committed:
Unittest Data_collection_class.py

UsernameAlexei commented on TCP_client.py:
```diff
@@ -0,0 +1,27 @@
+    time_now = datetime.now()
+    time_now = time_now.strftime("%H:%M:%f")
+
+    result = [time_now, f'{keystrokes.get_current_state()} press the key in 1 min']
```
Ready, updated.

UsernameAlexei commented on TCP_server.py:
```diff
@@ -0,0 +1,23 @@
+import socket
+
+sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Thank you.

UsernameAlexei commented on Data_collection_class.py:
```diff
@@ -0,0 +1,42 @@
+        self.counter -= 1
+        threading.Thread.join(self)
+
+# a = Collection()
```
Indeed, we must not forget about the tests. 
Uploaded test_collection.py

UsernameAlexei commented on TCP_server.py:
```diff
@@ -0,0 +1,23 @@
+        udata = data.decode()
+        client.send(udata.encode('utf-8'))
+        # save to file
+        with open('keystrokes.txt', 'a') as f:
```
Use a mutex?

arybin93 replied:
Yes, right

UsernameAlexei committed:
Update TCP_server.py

Update: 
+context manager

UsernameAlexei commented on TCP_server.py:
```diff
@@ -0,0 +1,23 @@
+import socket
+
+sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Updated

arybin93 commented on TCP_server.py:
```diff
@@ -0,0 +1,25 @@
+import ast
+from contextlib import closing
+
+with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
```
`with socket.socket() as s` should work as well

arybin93 reviewed (approved):
Looks better, thank you for your work. Homework is accepted.

UsernameAlexei committed:
Update Data_collection_class.py

undate:
handler

UsernameAlexei commented:
Thank you, the module is interesting, with clear examples in the lecture. Studying additional material.
